project/accounts/views.py

```diff
@@ -4,24 +4,43 @@
 This module will include views for the accounts app.
 """
 
-from core.custom_decorators import full_profile
+from accounts.authentication import send_activation_email
+from accounts.forms import ProfileEditForm, UserRegistrationForm
+from accounts.models import Profile
 from django.conf import settings
 from django.contrib.auth import get_user_model, login
 from django.contrib.auth import views as auth_views
+from django.contrib.auth.decorators import login_required
 from django.contrib.auth.mixins import LoginRequiredMixin
 from django.contrib.sites.shortcuts import get_current_site
 from django.http import HttpResponseRedirect
+from django.shortcuts import get_object_or_404, redirect
 from django.template.response import TemplateResponse
-from django.urls import reverse_lazy
-from django.utils.encoding import force_str
-from django.utils.http import urlsafe_base64_decode
-from django.utils.decorators import method_decorator
+from django.urls import reverse, reverse_lazy
 from django.views import View
 from django.views.generic.edit import FormView, UpdateView
 
-from accounts.authentication import account_activation_token, send_activation_email
-from accounts.forms import ProfileEditForm, UpdateProfileImage, UserRegistrationForm
-from accounts.models import Profile
+
+class ProfileFollow(LoginRequiredMixin, View):
+    def get(self, request, *args, **kwargs):
+        following_profile = Profile.objects.get(user__username=kwargs["username"])
+
+        self.request.user.profile.following.add(following_profile)
+
+        redirect_to = reverse("profile", kwargs={"username": kwargs["username"]})
+
+        return HttpResponseRedirect(redirect_to)
+
+
+class ProfileUnfollow(LoginRequiredMixin, View):
+    def get(self, request, *args, **kwargs):
+        following_profile = Profile.objects.get(user__username=kwargs["username"])
+
+        self.request.user.profile.following.remove(following_profile)
+
+        redirect_to = reverse("profile", kwargs={"username": kwargs["username"]})
+
+        return HttpResponseRedirect(redirect_to)
```

Comment:
A user should not follow or unfollow their own account. You can add an if statement to check it:

```python
if request.user.username === kwargs["username"]:
    return
```

Reply:
This is now fixed in both the view and template.